### ISSUE TITLE
plugin installs, shows icon but doesn't do anything 

### ISSUE THREAD

**Issue by degoya** [PTony/HTML5-one-click-checker#1]
i've installed in the actual firefox version, the icon shows up in my toolbar. when i click on it a window appears and show HTML5 one-click checker and a link to this repo.

**Comment by PTony** [comment]
Oops, I think I miss the notification for this issue. 
Is this always happening? Can you access http://validator.w3.org/nu/ ?

**Comment by mcka-dev** [comment]
I have also faced this problem.  when i click on it a window appears and show HTML5 one-click checker and a link to this repo. Access http://validator.w3.org/nu/ have.

**Comment by PTony** [comment]
Does it happen every time ?  
Is it a window or just a panel ?  
Can you take a screenshot and upload it here?  

Firefox extensions now have to use the new webextension API, so I will probably rewrite it later.

**Comment by mcka-dev** [comment]
It happens all the time.
![Icon without notifications](http://ipic.su/img/img7/fs/2017-07-1909_33_19-Amazon.1500446235.png)
![Only link to this repo](http://ipic.su/img/img7/fs/2017-07-1909_33_33-GraficheskijredaktorGreenshot.1500446355.png)
![Access validator.w3.org have](http://ipic.su/img/img7/fs/2017-07-1909_35_06-Showingresultsforhttps___www.1500446371.png)

**Comment by PTony** [comment]
For what I have tested (ubuntu 16.04 - firefox 54.0 - 64bits), It work half of the time, or display a "Err" black label like in screenshot below:  
![screenshot from 2017-07-19 09-20-35](https://user-images.githubusercontent.com/155288/28355456-5f8332a4-6c64-11e7-8d72-a48a4c1566dc.png)

I will test on windows 10.

**Comment by mcka-dev** [comment]
My screenshots are made on Windows 10

**Comment by PTony** [comment]
I can't reproduce. It work like for ubuntu for me (Windows 10 64bits - Firefox 54.0.1 32bits).

Can you confirm you are using HTML5 one-click checker v0.0.3?
Do you use others extensions? Can you tell me which ones?  
Can you create an new firefox profile with just HTML5-one-click-checker extension and retest?

**Comment by mcka-dev** [comment]
![2017-07-19 12_59_28-](https://user-images.githubusercontent.com/16173503/28361873-e6ff4d94-6c82-11e7-912c-69dae9e06272.png)
![2017-07-19 13_05_41-](https://user-images.githubusercontent.com/16173503/28361913-13481660-6c83-11e7-89ed-4deefe95d315.png)

**Comment by mcka-dev** [comment]
(Windows 7 32bits - Firefox 54.0.1 32bits).
![Uploading 2017-07-19 13_49_26-Windows 7 (Rus) x86 (IE8) [Работает] - Oracle VM VirtualBox.png…]()

**Comment by PTony** [comment]
I've installed firefox russian language and tested with same extensions disabled or not.
I can't reproduce the issue.  
Can you test with a new profile?

![addons](https://user-images.githubusercontent.com/155288/28364322-3f64d28c-6c84-11e7-8065-91fd43f09aeb.PNG)
![ff](https://user-images.githubusercontent.com/155288/28364326-4578a7d4-6c84-11e7-9e2a-e571249ccb00.PNG)

**Comment by mcka-dev** [comment]
Windows XP 32bits Rus - Firefox 52.2.1 32bits - HTML5 one-click checker v0.0.3 - Good!
![2017-07-19 14_42_51-winxp rus x86 - oracle vm virtualbox](https://user-images.githubusercontent.com/16173503/28365488-81fe25d2-6c91-11e7-976c-8e6eb25e0426.png)

Debian 8.8 64bits Eng - Firefox 52.2.0 (64-bit) - HTML5 one-click checker v0.0.3 - Good!
![2017-07-19 14_41_11-](https://user-images.githubusercontent.com/16173503/28365351-e757ec20-6c90-11e7-8256-f44862696643.png)

**Comment by PTony** [comment]
I think, I have to set some virtualboxes :)

**Comment by mcka-dev** [comment]
I created a new profile as shown [here](https://support.mozilla.org/en-US/kb/profile-manager-create-and-remove-firefox-profiles). Installed the add-on HTML5 one-click checker v0.0.3. In a new profile works!

Then I switched to the old profile here does not work.

**Comment by PTony** [comment]
Can you copy-paste the raw data of both profils so we can compare them. [Here the help page for copy the raw data](https://support.mozilla.org/en-US/kb/use-troubleshooting-information-page-fix-firefox)

**Comment by mcka-dev** [comment]
Yes, of course. [Here](https://www.dropbox.com/s/4gde9owwn66k480/two_profiles.zip?dl=0) is the archive of Windows 7 32bit VirtualBox, there are two profile: 
default - not works, 
andrei - works.

**Comment by PTony** [comment]
Thank you for your help. I will take a look at it a bit later and will come comment here.

**Comment by PTony** [comment]
Ok, I took a look at it:  

[diff of raw data of your 2 profiles (buggy one on left side)](http://www.mergely.com/kSXwn9Oq/)
There are 2 hidden extensions, "Follow-on Search Telemetry" and "Shield Recipe Client" (mozilla team extensions) which are not in the working profile, but maybe it's not related.

Also when I display the extension messages in console, i got this:
```
03:16:33,467 html5-one-click-checker:An error occurred during your request: 0:  sandbox.js:340
03:16:33,469 html5-one-click-checker:Error during the request  sandbox.js:340
```
which is generated by this code:
```javascript
  // formData to submit
  data = new FormData();
  data.append("content", src);
  data.append("docselect", "textarea");
  data.append("showsource", "no");

  var xhr = new XMLHttpRequest()
  xhr.timeout = 7000; // set the timeout to 7 sec
  xhr.ontimeout = function () {
    console.error("The request for " + url + " timed out.");
  };
  xhr.onerror = function () {
    console.error("Error during the request");
  };  
  xhr.onreadystatechange = function(){
    if(this.readyState === 4) {
      if(this.status === 200) { 
        self.port.emit("do-sanitizeHTML", this.responseText);

      } else {
        console.error('An error occurred during your request: ' +  this.status + ':' + this.statusText);
      } 
    }    
  }
  
  xhr.open("post", url)
  xhr.send(data);
```

It's a weird issue. At the moment I think it's better if people re-create a clean profile, and for me If I take time to work on it, to focus on porting it to the new firefox API.

Again, thanks for your help
